### ISSUE TITLE
options should supports date range

### ISSUE THREAD

**Issue by lielran** [eshaham/israeli-bank-scrapers#144]
currently, `options` object contains only `startDate`. 

given that credit cards (as far as I know) are working by months while bank transactions support date ranges, I'm missing the option to provide `endDate` or maybe spilt it to credit card vendors(input will be months e.g 07/2018) and date ranges for bank transactions(e.g. 01/01/2018 -
01/02/2018)

**Comment by esakal** [comment]
@lielran  
The common use-case for this library is to scrape all transactions starting a specific date. I assume you consume this library from your own project so you can optimize the results there. You can, for example ,scrape using `startdate` and then filter till `enddate` in your application.  

You mentioned that credit cards are working by months but it doesn't represent a regular month (e.g 1/6 - 30/6). In their site 06/2018 represent all transactions they will be billed sometime during this month. This is an information that we don't have so the scrapers will not reflect the same results as you see in the site. 

In general, There are many optimizations that can be done on the results. [israeli-ynab-updater](https://github.com/eshaham/israeli-ynab-updater/blob/master/src/scrape/scrape-individual.js#L108) for example allow filtering out future transactions or pending transactions.


wdyt?

**Comment by eshaham** [comment]
I can see a use case for which this can be useful - especially if someone uses this module to scrape data on behalf of other users, and would like to avoid fetching irrelevant data for the user.
However, since none of the banks allow for more than one year worth of transactions, I'm not sure how important is this feature, in terms of saving compute time. And like @esakal mentioned, this can easily be handled in the client.

**Comment by lielran** [comment]
but most of the banks support date range, right? 
why do we assuming the end day is today? over fetching can handle by the client for sure, but why should we pay the runtime overhead(all the extra data that is doing to be filtered).

my suggestion is to provide an optional end date where the default is none(=today). what do you say?

**Comment by esakal** [comment]
@lielran I reviewed all scrapers where they build the date range and this is what I came up with:

**hapoalim** - seems easy to implement
**leumi-card, isracard, amex** - they pass month range so if we will want to support end date range we will need to take extra month and then filter by range locally.
**otsar-hahayal** - it seems easy to implement but I cannot confirm it as I don't have an account there.
**visa-cal** - I think it is doable.
**discount** - this scraper creates a url to fetch the data. it contains [fromDate query param](https://github.com/eshaham/israeli-bank-scrapers/blob/master/src/scrapers/discount.js#L44). I assume it support also `toDate` but I cannot confirm it as I don't have an account there.

We will need to adjust all scrapers do support it so I'm not sure if it worth the effort but I don't mind join forces and extend some scrapers that I have access to if you both @eshaham and @lielran think it worth it. wdyt?
